### DISCUSSION TITLE
Cornering Time

### DISCUSSION THREAD

**Discussion by nicZox** [theOehrly/Fast-F1#111] [Q&A]
Hi everyone, how can I calculate the time it takes the driver to tackle a corner? I give an example:
 If I want to know the time it took the pilot to travel from 5000m to 6000m what can I do? Thanks

**Comment by theOehrly** [comment]
If you have a telemetry object with 'Distance' data, you can use pandas indexing to specifically select all parts where Distance >= 5000 and Distance <= 6000. If you then subtract the timestamps of the first sample from the last sample, you will get the time it took for the driver to drive this distance.

**Comment by Majd-ui** [comment]
Hi did you manage to do it, if yes can you give me a sample of your code

**Comment by markste-in** [comment]
Hope that helps... a little minimal example
```
#%% Imports
import fastf1 as ff1
import pandas as pd
ff1.Cache.enable_cache('cache')
event = ff1.get_session(2022, 1, 'FP1')
event.load()

#%% Get Data
tel = []
laps = event.laps
for lap in laps.pick_driver("HAM").iterlaps():
    df = lap[1].get_car_data()
    df["LapNumber"] = lap[1]["LapNumber"]
    tel.append(df.add_distance())
data = pd.concat(tel)

#%% Filter for just between 4000 and 4500 m
data = data[(data["Distance"]>4000)&(data["Distance"]<4500)]

#%% Taking Lap 2
specific_lap = data[data["LapNumber"]==2]
delta = specific_lap["Time"].max() - specific_lap["Time"].min()
print(pd.Timedelta.total_seconds(delta))
```
`8.961`
